packages/labs/react/src/create-component.ts

```diff
@@ -48,8 +48,13 @@ export type ReactWebComponent<
 // lifecycle methods or allow user to explicitly provide props.
 type ElementProps<I> = Partial<Omit<I, keyof HTMLElement>>;
 
-// Acceptable props to the React component.
-type ComponentProps<I, E extends EventNames = {}> = React.HTMLAttributes<I> &
+// Acceptable props to the React component. Omit keyof E from HTMLAttributes to
+// prefer provided event handler mapping over React's built-in event handler
+// props.
```

Comment:
Does this actually _prefer_ the provided types or just remove all the default event types? ie, does `onClick` still remain if it's not defined in the createComponent events?

Reply:
It'll only remove the default event type if a user provided an `events` option with a conflicting name like `onClick` which is captured in the `E` type. Otherwise, nothing will be omitted.

Reply:
oh right... sorry I was reading the type wrong!